parachain/pallets/omni-account/src/lib.rs

```diff
@@ -0,0 +1,187 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub use core_primitives::{Identity, MemberIdentity};
+pub use frame_system::pallet_prelude::BlockNumberFor;
+pub use pallet::*;
+
+use frame_support::pallet_prelude::*;
+use frame_system::pallet_prelude::*;
+use sp_core::H256;
+use sp_core_hashing::blake2_256;
+use sp_std::vec::Vec;
+
+pub trait AccountIdConverter<T: Config> {
+	fn convert(account_id: T::AccountId) -> Identity;
+}
```

Comment:
We can do `Identity::from(account_id)` if `account_id` is a substrate account but because at the pallet level rust does not know the concrete type of `T::AccountId`, we can't use it directly so I've added this as a workaround to get the substrate Identity.

Reply:
That is true, but only for the case of substrate.

I see that it's mainly used to compute `prime_identity_hash` - but what if the prime identity is btc/evm? So it means in `link_identity`, `who` should be of `Identity` type I guess?